File: scanlibrary/src/main/java/com/scanlibrary/AdaptiveAds.java
package com.scanlibrary;

import android.content.Context;
import android.util.DisplayMetrics;
import android.view.Display;
import android.view.WindowManager;

import com.google.android.gms.ads.AdSize;

class AdaptiveAds {
    Context context;

    AdaptiveAds(Context context) {
        this.context = context;
    }

    public AdSize getSize() {
        WindowManager window = (WindowManager)
                context.getSystemService(Context.WINDOW_SERVICE);

        Display display = window.getDefaultDisplay();
        DisplayMetrics outMetrics = new DisplayMetrics();
        display.getMetrics(outMetrics);
        float widthPixels = (float) outMetrics.widthPixels;
        float density = outMetrics.density;
        int adWidth = (int) (widthPixels / density);
        return AdSize.getCurrentOrientationAnchoredAdaptiveBannerAdSize(
                context,
                adWidth
        );
    }


}

File: app/src/main/java/com/pakthemekeyboard/easyurdukeyboard/pdfreader/utils/ads/AdaptiveAds.kt
package com.pakthemekeyboard.easyurdukeyboard.pdfreader.utils.ads

import android.content.Context
import android.util.DisplayMetrics
import android.view.WindowManager
import com.google.android.gms.ads.AdSize

class AdaptiveAds(private var contextA: Context) {
    val adSize: AdSize
        get() {
            val window =
                (contextA.getSystemService(Context.WINDOW_SERVICE) as WindowManager)
            val display = window.defaultDisplay
            val outMetrics = DisplayMetrics()
            display.getMetrics(outMetrics)
            val widthPixels = outMetrics.widthPixels.toFloat()
            val density = outMetrics.density
            val adWidth = (widthPixels / density).toInt()
            return AdSize.getCurrentOrientationAnchoredAdaptiveBannerAdSize(
                contextA,
                adWidth
            )
        }

}